core/rawdb/accessors_trie.go

```diff
@@ -335,7 +357,12 @@ func ParseStateScheme(provided string, disk ethdb.Database) (string, error) {
 	// If state scheme is not specified, use the scheme consistent
 	// with persistent state, or fallback to hash mode if database
 	// is empty.
-	stored := ReadStateScheme(disk)
+	var stored string
+	if disk != nil && disk.StateStore() != nil {
```

Comment:
Simplify the code.

```
stored := ReadStateScheme(disk)
```

Use disk.StateStore inside the `ReadStateScheme`.

Reply:
fixed